libs/@guardian/cobalt-plugin-ts/index.mjs

```diff
@@ -0,0 +1,92 @@
+// @ts-check
+
+import { defaultTransformer } from '@cobalt-ui/plugin-js';
+import { set } from '@cobalt-ui/utils';
+import ts from 'typescript';
+
+/**
+ * @param {{ filename: string; }} options
+ * @returns {import('@cobalt-ui/core').Plugin}
+ */
+export default function pluginTS(options) {
+	return {
+		name: 'plugin-ts',
+		config(config) {},
+		async build({ tokens, metadata, rawSchema }) {
+			const header = `/**
+* Design Tokens
+* Autogenerated from tokens.json.
+* DO NOT EDIT!
+*/
+
+`;
+
+			// this is where we'll store the transformed tokens
+			const transformedTokens = {};
+
+			// we can re-use the default transformer from `@cobalt-ui/plugin-js`
+			for (const token of tokens) {
+				set(transformedTokens, token.id, defaultTransformer(token));
+			}
+
+			// create a typescript source string containing the transformed
+			// tokens
+			const typescriptSource = `export const tokens = ${JSON.stringify(transformedTokens, null, 2)} as const;`;
+
+			// now we create a ts program to create the JS and declaration file
+			// contents. the program would write to disk by default, so we
+			// override that behaviour to capture the output in memory instead.
+
+			// this is where we'll store the output for the JS and declaration
+			// files
+			const output = {};
+
+			const program = ts.createProgram(
+				['fakefile.ts'],
```

Comment:
Would extracting this to a const help with maintainability? Maybe even assign it to a `Symbol`